flanker/addresslib/address.py

```diff
@@ -558,7 +558,7 @@ def to_unicode(self):
         Converts to unicode.
         """
         if self.display_name:
-            return u'{} <{}@{}>'.format(self.display_name, self.local_part, self.domain)
+            return u'{} <{}@{}>'.format(smart_quote(self.display_name), self.local_part, self.domain)
```

Comment:
If I am not mistaken `smart_quote` turns display name with at least one utf-8 character to something like `=?utf-8?b?0LDQtNC20LDQuQ==?=`. Although this solution works, it would be great if the result would be a readable name, where only characters that impede parsing are quoted. Is that possible?